debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/FindModuleByNameCommand.java

```diff
@@ -0,0 +1,109 @@
+ *******************************************************************************/
+package com.ibm.j9ddr.vm29.tools.ddrinteractive.commands;
+
+import java.io.PrintStream;
+
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.tools.ddrinteractive.Command;
+import com.ibm.j9ddr.tools.ddrinteractive.Context;
+import com.ibm.j9ddr.tools.ddrinteractive.DDRInteractiveCommandException;
+import com.ibm.j9ddr.util.PatternString;
+import com.ibm.j9ddr.vm29.j9.DataType;
+import com.ibm.j9ddr.vm29.j9.ObjectModel;
+import com.ibm.j9ddr.vm29.j9.Pool;
+import com.ibm.j9ddr.vm29.j9.SlotIterator;
+import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ModulePointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ObjectPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9PoolPointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+import com.ibm.j9ddr.vm29.pointer.helper.J9RASHelper;
+import com.ibm.j9ddr.vm29.tools.ddrinteractive.JavaVersionHelper;
+
+/**
+ * FindModuleByName command find the modules corresponding to its module name
+ * 
+ * Example:
+ * 			!findmodulebyname java.base
+ * Example output: 
+ * 			Searching for modules named 'java.base' in VM=13053677e00
+ *			java.base			!j9module 0x00000130550C7E88
+ *			Found 1 module(s) named java.base
+ */
+public class FindModuleByNameCommand extends Command
+{
+	public FindModuleByNameCommand()
+	{
+		addCommand("findmodulebyname", "<moduleName>", "find the modules corresponding to its module name");
+	}
+
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException
+	{
+		if (args.length == 0) {
+			printUsage(out);
+			return;
+		}
+		try {
+			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+			if (JavaVersionHelper.ensureJava9AndUp(vm, out)) {
+				J9PoolPointer pool = vm.modularityPool();
+				Pool<J9ModulePointer> modulePool = Pool.fromJ9Pool(pool, J9ModulePointer.class);
+				SlotIterator<J9ModulePointer> poolIterator = modulePool.iterator();
+				J9ModulePointer modulePtr = null;
+
+				int hitCount = 0;
+				String searchModuleName = args[0];
+				J9ObjectPointer moduleNameObjPtr = null;
+				PatternString pattern = new PatternString(searchModuleName);
+
+				out.println(String.format("Searching for modules named '%s' in VM=%s", searchModuleName,
+						Long.toHexString(vm.getAddress())));
+				while (poolIterator.hasNext()) {
+					modulePtr = poolIterator.next();
+					moduleNameObjPtr = modulePtr.moduleName();
+					if (ObjectModel.isPointerInHeap(vm, moduleNameObjPtr)
+							&& pattern.isMatch(J9ObjectHelper.stringValue(moduleNameObjPtr))) {
+						hitCount++;
+
+						String moduleName = J9ObjectHelper.stringValue(moduleNameObjPtr);
+						String hexAddress = modulePtr.getHexAddress();
+						out.printf("%-30s!j9module %s\n", moduleName, hexAddress);
```

Comment:
There should be at least one space between `%-30s` and `!j9module` so a module name of 30 characters or more is still separated from the command. You should also use `%n` instead of `\n`.